src/dso_api/dynamic_api/views/mvt.py

```diff
@@ -0,0 +1,96 @@
+"""Mapbox Vector Tiles (MVT) views of geographic datasets."""
+
+from typing import Tuple
+
+from django.contrib.gis.db.models import GeometryField
+from django.core.exceptions import FieldDoesNotExist, PermissionDenied
+from django.http import Http404
+from django.views.generic import TemplateView
+from schematools.contrib.django.models import Dataset
+from schematools.utils import to_snake_case, toCamelCase
+from vectortiles.postgis.views import MVTView
+
+from dso_api.dynamic_api import permissions
+
+
+class DatasetMVTIndexView(TemplateView):
+    """Overview of available MVT endpoints."""
+
+    template_name = "dso_api/dynamic_api/mvt_index.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        from ..urls import router
+
+        dataset_names = set(router.all_models.keys())
+        datasets = (
+            (ds.name, list(self._have_geometry_fields(ds.schema)), ds.schema)
+            for ds in Dataset.objects.db_enabled().order_by("name")
+            if ds.name in dataset_names
+        )
+        context["datasets"] = [
+            (name, fields, schema) for name, fields, schema in datasets if fields
+        ]
+
+        return context
+
+    def _have_geometry_fields(self, schema):
+        """Yields names of tables that have a geometry field."""
+        for table in schema["tables"]:
+            for field in table["schema"]["properties"].values():
+                ref_type = field.get("$ref")
+                if ref_type and ref_type.startswith("https://geojson.org/schema/"):
+                    yield to_snake_case(table["id"])
+
+
+class DatasetMVTView(MVTView):
+    """An MVT view for a single dataset."""
+
+    permission_classes = [permissions.HasOAuth2Scopes]
+
+    def setup(self, request, *args, **kwargs):
+        super().setup(request, *args, **kwargs)
+        from ..urls import router
+
+        dataset_name = self.kwargs["dataset_name"]
+        table_name = self.kwargs["table_name"]
+
+        try:
+            model = router.all_models[dataset_name][table_name]
+            self._unauthorized = permissions.get_unauthorized_fields(request, model)
+            self.model = model
+            self.check_permissions(request)
+        except KeyError:
+            raise Http404("Invalid dataset") from None
+
+    def get(self, request, *args, **kwargs):
+        kwargs.pop("dataset_name")
+        kwargs.pop("table_name")
+        return super().get(request, *args, **kwargs)
+
+    def get_queryset(self):
+        return self.model.objects.filter()
```

Comment:
```suggestion
        return self.model.objects.all()
```

Een lege `filter()` is niet echt handig, dan liever de default `.all()`.